File: app/src/main/java/com/example/ui/screens/ProfileScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.GlassCard
import com.example.ui.components.NeonButton
import com.example.ui.theme.*
import com.example.viewmodel.QuizoraViewModel

@Composable
fun ProfileScreen(
    viewModel: QuizoraViewModel,
    onSwitchLanguage: () -> Unit
) {
    val progress by viewModel.userProgress.collectAsState()
    val activeClass by viewModel.currentClass.collectAsState()

    val badgesList = remember(progress.badgesUnlocked) {
        val unlocked = progress.badgesUnlocked.split(",").map { it.trim() }.filter { it.isNotEmpty() }.toSet()
        listOf(
            BadgeDetail("Quiz Rookie", "Awarded upon entering Quizora in Guest Mode.", "WorkspacePremium", unlocked.contains("Quiz Rookie")),
            BadgeDetail("Quiz King", "Reach 200 XP to master speed challenges.", "EmojiEvents", unlocked.contains("Quiz King")),
            BadgeDetail("Science Master", "Unlock 350 XP to claims the science scepter.", "Science", unlocked.contains("Science Master")),
            BadgeDetail("Math Genius", "Unlock 3 or more total badges successfully.", "Functions", unlocked.contains("Math Genius")),
            BadgeDetail("CBSE Champion", "Earn 500 XP to achieve the board ranking.", "LocalActivity", unlocked.contains("CBSE Champion"))
        )
    }

    LazyColumn(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = 20.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp),
        contentPadding = PaddingValues(top = 24.dp, bottom = 24.dp)
    ) {
        // Upper Gamer Avatar Badge Card
        item {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(GlassBg, RoundedCornerShape(24.dp))
                    .border(1.dp, NeonBlue.copy(alpha = 0.5f), RoundedCornerShape(24.dp))
                    .padding(24.dp),
                contentAlignment = Alignment.Center
            ) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Box(
                        contentAlignment = Alignment.Center,
                        modifier = Modifier
                            .size(76.dp)
                            .background(NeonPurple.copy(alpha = 0.2f), shape = CircleShape)
                            .border(1.5.dp, NeonPurple, CircleShape)
                    ) {
                        Icon(
                            imageVector = Icons.Default.AccountCircle,
                            contentDescription = "Guest Student Avatar",
                            tint = NeonPurple,
                            modifier = Modifier.size(52.dp)
                        )
                    }

                    Spacer(modifier = Modifier.height(14.dp))

                    Text(
                        text = "QUIZORA GUEST",
                        fontSize = 18.sp,
                        fontWeight = FontWeight.Black,
                        color = Color.White,
                        letterSpacing = 1.sp
                    )

                    Text(
                        text = "CBSE ${activeClass.uppercase()} STUDENT",
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Bold,
                        color = NeonBlue,
                        letterSpacing = 1.sp
                    )
                }
            }
        }

        // Stats Display Block
        item {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                GamerStatBox(
                    modifier = Modifier.weight(1f),
                    title = "TOTAL XP",
                    value = "${progress.xp} XP",
                    color = NeonBlue,
                    icon = Icons.Default.Star
                )
                GamerStatBox(
                    modifier = Modifier.weight(1f),
                    title = "DAILY STREAK",
                    value = "${progress.streak} Days",
                    color = GoldStreak,
                    icon = Icons.Default.LocalFireDepartment
                )
            }
        }

        // Language / Class Quick Selectors
        item {
            GlassCard(borderGlowColor = NeonPurple) {
                Text(
                    text = "ACADEMICS SETTINGS",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Bold,
                    color = NeonPurple,
                    letterSpacing = 1.sp
                )
                Spacer(modifier = Modifier.height(14.dp))

                SettingsInteractiveRow(
                    title = "Instruction Medium",
                    subtitle = progress.selectedLanguage.ifEmpty { "English" },
                    icon = Icons.Default.Translate,
                    onClick = onSwitchLanguage
                )

                Spacer(modifier = Modifier.height(10.dp))
                Divider(color = TextMuted.copy(alpha = 0.2f))
                Spacer(modifier = Modifier.height(10.dp))

                SettingsInteractiveRow(
                    title = "Selected Standard",
                    subtitle = activeClass,
                    icon = Icons.Default.School,
                    onClick = { /* Standard changing triggered via home screens */ }
                )
            }
        }

        // Gamer Badges Checklist
        item {
            Text(
                text = "BADGES PROGRESSION",
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                color = TextSecondary,
                letterSpacing = 1.sp,
                modifier = Modifier.padding(top = 6.dp)
            )
        }

        items(badgesList) { badge ->
            BadgeRowCard(badge = badge)
        }

        // Reset statistics trigger for easy testing
        item {
            Column(
                modifier = Modifier.fillMaxWidth().padding(top = 10.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text(
                    text = "Developer Diagnostics Mode (Guest Sandbox)",
                    fontSize = 11.sp,
                    color = TextMuted
                )
                Spacer(modifier = Modifier.height(8.dp))
                NeonButton(
                    text = "RESET ALL STATISTICS",
                    onClick = { viewModel.resetAllProgress() },
                    modifier = Modifier.fillMaxWidth().testTag("reset_progress_button"),
                    isPrimary = false,
                    glowColor = NeonRed
                )
            }
        }
    }
}

data class BadgeDetail(
    val title: String,
    val description: String,
    val iconName: String,
    val isUnlocked: Boolean
)

@Composable
fun GamerStatBox(
    modifier: Modifier = Modifier,
    title: String,
    value: String,
    color: Color,
    icon: androidx.compose.ui.graphics.vector.ImageVector
) {
    Box(
        modifier = modifier
            .background(GlassBg, RoundedCornerShape(16.dp))
            .border(0.5.dp, color.copy(alpha = 0.3f), RoundedCornerShape(16.dp))
            .padding(16.dp)
    ) {
        Column {
            Icon(imageVector = icon, contentDescription = null, tint = color, modifier = Modifier.size(20.dp))
            Spacer(modifier = Modifier.height(8.dp))
            Text(text = title, fontSize = 10.sp, color = TextSecondary, fontWeight = FontWeight.Bold)
            Text(text = value, fontSize = 20.sp, color = Color.White, fontWeight = FontWeight.Bold)
        }
    }
}

@Composable
fun SettingsInteractiveRow(
    title: String,
    subtitle: String,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    onClick: () -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(vertical = 4.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(10.dp)) {
            Icon(imageVector = icon, contentDescription = title, tint = TextSecondary, modifier = Modifier.size(20.dp))
            Column {
                Text(text = title, fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color.White)
                Text(text = subtitle, fontSize = 12.sp, color = TextMuted)
            }
        }
        Icon(imageVector = Icons.Default.ChevronRight, contentDescription = null, tint = TextMuted, modifier = Modifier.size(20.dp))
    }
}

@Composable
fun BadgeRowCard(badge: BadgeDetail) {
    val borderColor = if (badge.isUnlocked) NeonPurple.copy(alpha = 0.6f) else TextMuted.copy(alpha = 0.2f)
    
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .background(GlassBg, RoundedCornerShape(16.dp))
            .border(0.5.dp, borderColor, RoundedCornerShape(16.dp))
            .padding(16.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(12.dp),
                modifier = Modifier.weight(1f)
            ) {
                Box(
                    contentAlignment = Alignment.Center,
                    modifier = Modifier
                        .size(42.dp)
                        .background(if (badge.isUnlocked) NeonPurple.copy(alpha = 0.2f) else Color(0x10FFFFFF), shape = CircleShape)
                        .border(1.dp, if (badge.isUnlocked) NeonPurple else TextMuted, CircleShape)
                ) {
                    val icon = when (badge.iconName) {
                        "EmojiEvents" -> Icons.Default.EmojiEvents
                        "Science" -> Icons.Default.Science
                        "Functions" -> Icons.Default.Functions
                        "LocalActivity" -> Icons.Default.LocalActivity
                        else -> Icons.Default.WorkspacePremium
                    }
                    Icon(
                        imageVector = icon,
                        contentDescription = badge.title,
                        tint = if (badge.isUnlocked) GoldStreak else TextMuted,
                        modifier = Modifier.size(22.dp)
                    )
                }

                Column {
                    Text(
                        text = badge.title,
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Bold,
                        color = if (badge.isUnlocked) Color.White else TextSecondary
                    )
                    Text(
                        text = badge.description,
                        fontSize = 11.sp,
                        color = TextMuted,
                        lineHeight = 14.sp
                    )
                }
            }

            if (badge.isUnlocked) {
                Icon(
                    imageVector = Icons.Default.CheckCircle,
                    contentDescription = "Unlocked badge icon indicator",
                    tint = NeonGreen,
                    modifier = Modifier.size(20.dp)
                )
            } else {
                Icon(
                    imageVector = Icons.Default.Lock,
                    contentDescription = "Locked badge",
                    tint = TextMuted,
                    modifier = Modifier.size(16.dp)
                )
            }
        }
    }
}

File: lib/screens/profile_screen.dart
import 'package:flutter/material.dart';
import '../components/quizora_components.dart';
import '../data/database_helper.dart';
import '../models/user_progress.dart';

class BadgeDetail {
  final String title;
  final String description;
  final IconData icon;
  final bool isUnlocked;

  BadgeDetail(this.title, this.description, this.icon, this.isUnlocked);
}

class ProfileScreen extends StatefulWidget {
  final UserProgress progress;
  final VoidCallback onSwitchLanguage;
  final VoidCallback onRefreshProgress;

  const ProfileScreen({
    Key? key,
    required this.progress,
    required this.onSwitchLanguage,
    required this.onRefreshProgress,
  }) : super(key: key);

  @override
  _ProfileScreenState createState() => _ProfileScreenState();
}

class _ProfileScreenState extends State<ProfileScreen> {
  @override
  Widget build(BuildContext context) {
    final unlockedBadges = widget.progress.badgesUnlocked
        .split(',')
        .map((e) => e.trim())
        .where((e) => e.isNotEmpty)
        .toSet();

    final List<BadgeDetail> badgesList = [
      BadgeDetail('Quiz Rookie', 'Awarded upon entering Quizora in Guest Mode.', Icons.workspace_premium, unlockedBadges.contains('Quiz Rookie')),
      BadgeDetail('Quiz King', 'Reach 200 XP to master speed challenges.', Icons.emoji_events, widget.progress.xp >= 200),
      BadgeDetail('Science Master', 'Unlock 350 XP to claim the science scepter.', Icons.science, widget.progress.xp >= 350),
      BadgeDetail('Math Genius', 'Unlock 3 or more total badges successfully.', Icons.functions, unlockedBadges.length >= 3),
      BadgeDetail('CBSE Champion', 'Earn 500 XP to achieve the board ranking.', Icons.local_activity, widget.progress.xp >= 500),
    ];

    return ListView(
      padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 24.0),
      children: [
        // Upper Gamer Avatar Badge Card
        Container(
          width: double.infinity,
          decoration: BoxDecoration(
            color: glassBg,
            borderRadius: BorderRadius.circular(24.0),
            border: Border.all(color: neonBlue.withOpacity(0.5), width: 1.0),
          ),
          padding: const EdgeInsets.all(24.0),
          child: Column(
            children: [
              Container(
                width: 76,
                height: 76,
                decoration: BoxDecoration(
                  shape: BoxShape.circle,
                  color: neonPurple.withOpacity(0.2),
                  border: Border.all(color: neonPurple, width: 1.5),
                ),
                child: const Icon(
                  Icons.account_circle,
                  color: neonPurple,
                  size: 52,
                ),
              ),
              const SizedBox(height: 14),
              const Text(
                'QUIZORA GUEST',
                style: TextStyle(
                  color: Colors.white,
                  fontSize: 18,
                  fontWeight: FontWeight.black,
                  letterSpacing: 1.0,
                ),
              ),
              const SizedBox(height: 2),
              Text(
                'CBSE ${widget.progress.selectedClass.toUpperCase()} STUDENT',
                style: const TextStyle(
                  color: neonBlue,
                  fontSize: 11,
                  fontWeight: FontWeight.bold,
                  letterSpacing: 1.0,
                ),
              ),
            ],
          ),
        ),
        const SizedBox(height: 16),

        // Stats Display Block
        Row(
          children: [
            Expanded(
              child: _GamerStatBox(
                title: 'TOTAL XP',
                value: '${widget.progress.xp} XP',
                color: neonBlue,
                icon: Icons.star,
              ),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: _GamerStatBox(
                title: 'STREAK',
                value: '${widget.progress.streak} Days',
                color: Colors.orange,
                icon: Icons.local_fire_department,
              ),
            ),
          ],
        ),
        const SizedBox(height: 16),

        // Language / Class Academic Settings
        GlassCard(
          borderGlowColor: neonPurple,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              const Text(
                'ACADEMICS SETTINGS',
                style: TextStyle(
                  color: neonPurple,
                  fontSize: 11,
                  fontWeight: FontWeight.bold,
                  letterSpacing: 1.0,
                ),
              ),
              const SizedBox(height: 14),
              _SettingsInteractiveRow(
                title: 'Instruction Medium',
                subtitle: widget.progress.selectedLanguage.isEmpty ? 'English' : widget.progress.selectedLanguage,
                icon: Icons.translate,
                onClick: widget.onSwitchLanguage,
              ),
              const Divider(color: textMuted, height: 24),
              _SettingsInteractiveRow(
                title: 'Target Syllabus Class',
                subtitle: widget.progress.selectedClass,
                icon: Icons.play_for_work,
                onClick: () async {
                  // Switch class cycled
                  final currentClass = widget.progress.selectedClass;
                  final List<String> classes = ['Class 6', 'Class 7', 'Class 8', 'Class 9', 'Class 10', 'Class 11', 'Class 12'];
                  final index = classes.indexOf(currentClass);
                  final nextClass = classes[(index + 1) % classes.length];

                  final updated = widget.progress.copyWith(selectedClass: nextClass);
                  await DatabaseHelper.instance.saveProgress(updated);
                  widget.onRefreshProgress();
                },
              ),
            ],
          ),
        ),
        const SizedBox(height: 16),

        // Badges Progress
        const Text(
          'BADGES INVENTORY',
          style: TextStyle(
            color: textSecondary,
            fontSize: 13,
            fontWeight: FontWeight.bold,
            letterSpacing: 0.5,
          ),
        ),
        const SizedBox(height: 10),
        ...badgesList.map((badge) {
          return Container(
            margin: const EdgeInsets.only(bottom: 10.0),
            padding: const EdgeInsets.all(12.0),
            decoration: BoxDecoration(
              color: glassBg,
              borderRadius: BorderRadius.circular(16),
              border: Border.all(
                color: badge.isUnlocked ? neonPurple.withOpacity(0.5) : textMuted.withOpacity(0.15),
                width: 1.0,
              ),
            ),
            child: Row(
              children: [
                Opacity(
                  opacity: badge.isUnlocked ? 1.0 : 0.4,
                  child: Container(
                    width: 44,
                    height: 44,
                    decoration: BoxDecoration(
                      color: neonPurple.withOpacity(0.15),
                      shape: BoxShape.circle,
                    ),
                    child: Icon(badge.icon, color: badge.isUnlocked ? Colors.orange : textMuted, size: 22),
                  ),
                ),
                const SizedBox(width: 14),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        badge.title,
                        style: TextStyle(
                          color: badge.isUnlocked ? Colors.white : textSecondary,
                          fontWeight: FontWeight.bold,
                          fontSize: 14,
                        ),
                      ),
                      Text(
                        badge.description,
                        style: const TextStyle(color: textSecondary, fontSize: 11, height: 1.2),
                      ),
                    ],
                  ),
                ),
                if (!badge.isUnlocked)
                  const Icon(Icons.lock, color: textMuted, size: 16)
                else
                  const Icon(Icons.check_circle, color: Colors.greenAccent, size: 18),
              ],
            ),
          );
        }),
        const SizedBox(height: 30),
      ],
    );
  }
}

class _GamerStatBox extends StatelessWidget {
  final String title;
  final String value;
  final Color color;
  final IconData icon;

  const _GamerStatBox({
    Key? key,
    required this.title,
    required this.value,
    required this.color,
    required this.icon,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      decoration: BoxDecoration(
        color: glassBg,
        borderRadius: BorderRadius.circular(16.0),
        border: Border.all(color: color.withOpacity(0.35), width: 1.0),
      ),
      padding: const EdgeInsets.all(16.0),
      child: Row(
        children: [
          Icon(icon, color: color, size: 24),
          const SizedBox(width: 12),
          Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(
                title,
                style: const TextStyle(color: textSecondary, fontSize: 10, fontWeight: FontWeight.bold),
              ),
              Text(
                value,
                style: const TextStyle(color: Colors.white, fontSize: 15, fontWeight: FontWeight.bold),
              ),
            ],
          ),
        ],
      ),
    );
  }
}

class _SettingsInteractiveRow extends StatelessWidget {
  final String title;
  final String subtitle;
  final IconData icon;
  final VoidCallback onClick;

  const _SettingsInteractiveRow({
    Key? key,
    required this.title,
    required this.subtitle,
    required this.icon,
    required this.onClick,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: onClick,
      borderRadius: BorderRadius.circular(8.0),
      child: Padding(
        padding: const EdgeInsets.symmetric(vertical: 4.0),
        child: Row(
          children: [
            Icon(icon, color: neonBlue, size: 20),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    title,
                    style: const TextStyle(color: Colors.white, fontSize: 13, fontWeight: FontWeight.bold),
                  ),
                  Text(
                    subtitle,
                    style: const TextStyle(color: neonPurple, fontSize: 12, fontWeight: FontWeight.bold),
                  ),
                ],
              ),
            ),
            const Icon(Icons.edit, color: textSecondary, size: 16),
          ],
        ),
      ),
    );
  }
}
